test/e2e/test_attachment.py

```diff
@@ -136,11 +137,60 @@ def test_create_with_invalid_entity_id(self):
         self.check_post_attachment_failed_with_detail(422, "Invalid `entity_id` given")
 
 
-class ListDSL(CreateDSL):
-    """Base class for list tests."""
+class GetDSL(CreateDSL):
+    """Base class for get tests."""
 
     _get_response_attachment: Response
 
+    def get_attachment(self, attachment_id: str) -> None:
+        """
+        Gets an attachment with the given ID.
+
+        :param attachment_id: The ID of the attachment to be obtained.
+        """
+        self._get_response_attachment = self.test_client.get(f"/attachments/{attachment_id}")
+
+    def check_get_attachment_success(self, expected_attachment_data: dict) -> None:
+        """
+        Checks that a prior call to `get_attachment` gave a successful response with the expected data returned.
+
+        :param expected_attachment_data: Dictionary containing the expected attachment data as would be required
+            for an `AttachmentMetadataSchema`.
+        """
+        assert self._get_response_attachment.status_code == 200
+        assert self._get_response_attachment.json() == expected_attachment_data
+
+    def check_get_attachment_failed(self) -> None:
```

Comment:
```suggestion
    def check_get_attachment_failed(self, status_code: int, detail: str) -> None:
```
Could this match IMS and take in the message instead? I think this makes it a bit clearer what the type of failure is in the test itself and is easier to add more in future. Could you do the same to the corresponding image one as well please.